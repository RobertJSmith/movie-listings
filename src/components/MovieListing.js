import React, { Component } from 'react';

class MovieListing extends Component {
    constructor(props) {
        super(props);

        this.state = {
            id: props.id,
            imagePath: props.imgPath,
            title: props.title,
            popularityRating: props.rating,
            genresDisplay: props.genresDisplay
        }
    }

    render() {
        //I've left in the poplarityRating property here so that we can see the filtering is working correctly
        return (
            <div className="movie-listing-card">
                <img src={this.state.imagePath} alt={this.state.title}></img>
                <div className="movie-listing-card-info"> 
                    <h1 className="movie-listing-card-title">{this.state.title}</h1>
                    <p className="movie-listing-card-rating">Rating: {this.state.popularityRating}</p>
                    <p className="movie-listing-card-genres">{this.state.genresDisplay.toString()}</p>
                </div>
            </div>
        );
    }
}

export default MovieListing;