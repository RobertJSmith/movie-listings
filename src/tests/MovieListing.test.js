import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MovieListing from '../components/MovieListing';

configure({ adapter: new Adapter() });

test('should render MovieListing with test data', () => {
    /*This test is to make sure that we get the expected html output given the input we've defined.
    modifying the MovieListing render function return statement should then fail this test (as the rendered HTML would no longer
    match the snapshot stored (at least until we give the go ahead to override the snapshot)*/

    const props = {
        id: 299537,
        title: "Captain Marvel",
        rating: 7.3,
        imgPath: "https://image.tmdb.org/t/p/w300/AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg",
        genresDisplay: ["Action", "Adventure", "Science Fiction"]
    };
    
    const wrapper = shallow(<MovieListing {...props}/>);
    expect(wrapper).toMatchSnapshot();
});