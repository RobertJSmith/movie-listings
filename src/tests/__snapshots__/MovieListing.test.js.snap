// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieListing with test data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieListing
    genresDisplay={
      Array [
        "Action",
        "Adventure",
        "Science Fiction",
      ]
    }
    id={299537}
    imgPath="https://image.tmdb.org/t/p/w300/AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg"
    rating={7.3}
    title="Captain Marvel"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Captain Marvel"
          src="https://image.tmdb.org/t/p/w300/AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg"
        />,
        <div
          className="movie-listing-card-info"
        >
          <h1
            className="movie-listing-card-title"
          >
            Captain Marvel
          </h1>
          <p
            className="movie-listing-card-rating"
          >
            Rating: 
            7.3
          </p>
          <p
            className="movie-listing-card-genres"
          >
            Action,Adventure,Science Fiction
          </p>
        </div>,
      ],
      "className": "movie-listing-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Captain Marvel",
          "src": "https://image.tmdb.org/t/p/w300/AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="movie-listing-card-title"
            >
              Captain Marvel
            </h1>,
            <p
              className="movie-listing-card-rating"
            >
              Rating: 
              7.3
            </p>,
            <p
              className="movie-listing-card-genres"
            >
              Action,Adventure,Science Fiction
            </p>,
          ],
          "className": "movie-listing-card-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Captain Marvel",
              "className": "movie-listing-card-title",
            },
            "ref": null,
            "rendered": "Captain Marvel",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Rating: ",
                7.3,
              ],
              "className": "movie-listing-card-rating",
            },
            "ref": null,
            "rendered": Array [
              "Rating: ",
              7.3,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Action,Adventure,Science Fiction",
              "className": "movie-listing-card-genres",
            },
            "ref": null,
            "rendered": "Action,Adventure,Science Fiction",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Captain Marvel"
            src="https://image.tmdb.org/t/p/w300/AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg"
          />,
          <div
            className="movie-listing-card-info"
          >
            <h1
              className="movie-listing-card-title"
            >
              Captain Marvel
            </h1>
            <p
              className="movie-listing-card-rating"
            >
              Rating: 
              7.3
            </p>
            <p
              className="movie-listing-card-genres"
            >
              Action,Adventure,Science Fiction
            </p>
          </div>,
        ],
        "className": "movie-listing-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Captain Marvel",
            "src": "https://image.tmdb.org/t/p/w300/AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="movie-listing-card-title"
              >
                Captain Marvel
              </h1>,
              <p
                className="movie-listing-card-rating"
              >
                Rating: 
                7.3
              </p>,
              <p
                className="movie-listing-card-genres"
              >
                Action,Adventure,Science Fiction
              </p>,
            ],
            "className": "movie-listing-card-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Captain Marvel",
                "className": "movie-listing-card-title",
              },
              "ref": null,
              "rendered": "Captain Marvel",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Rating: ",
                  7.3,
                ],
                "className": "movie-listing-card-rating",
              },
              "ref": null,
              "rendered": Array [
                "Rating: ",
                7.3,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Action,Adventure,Science Fiction",
                "className": "movie-listing-card-genres",
              },
              "ref": null,
              "rendered": "Action,Adventure,Science Fiction",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
